Add DeliveryCreateNoteCommand implementation to DeveloperGuide.md

B-enguin opened the pull request:
Explained the current implementation of the DeliveryCreateNoteCommand in the DeveloperGuide Add activity and sequence diagrams for DeliveryCreateNoteCommand to support the explanation

Update DeliveryCreateNoteCommand.java and DeliveryStatusCommand.java to use `Model#getDelivery()` in accordance with Law of Demeter

Close #163

B-enguin committed:
Add DeliveryCreateNoteCommand implementation to DeveloperGuide.md

Explained the current implementation of the DeliveryCreateNoteCommand in the DeveloperGuide
Add activity and sequence diagrams for DeliveryCreateNoteCommand to support the explanation

Update DeliveryCreateNoteCommand.java and DeliveryStatusCommand.java to use `Model#getDelivery()` in accordance with Law of Demeter

zhonghan721 commented on docs/DeveloperGuide.md:
```diff
@@ -193,6 +193,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+5. `LogicManager` calls `DeliveryCreateNoteCommand#execute()`, first checking if the user is logged in by calling
+   `Model#getUserLoginStatus()`
+6. It thens attempt to fetch the `Delivery` with the specified `DELIVERY_ID`, and replaces that `Delivery` using
+   `Model#setDelivery()` with a newly created `Delivery` with identical fields except for its status which is
```
Should it be note instead of status?

zhonghan721 commented on docs/DeveloperGuide.md:
```diff
@@ -193,6 +193,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+4. If the user is not logged in during command execution, a `CommandException` will be thrown.
+5. The number provided is used to fetch the associated `Delivery` from `Model` if it exists. If the provided
+   number does not match any of the IDs of the `Delivery`s stored in `Model`, a `CommandException` is thrown.
+6. If the command completed successfully, the selected `Delivery` will be replaced by an identical `Delivery`
```
Would "completes" be more suitable here?

zhonghan721 commented on docs/diagrams/DeliveryCreateNoteActivityDiagram.puml:
```diff
@@ -0,0 +1,50 @@
+    else ([Not Null])
+    endif
+    : Tokenize Arguments with Note prefix;
+    if (Check Prefix Present\nand Preamble Empty) then ([Prefix Not Present or Preamble not Empty])
+        :ParseException: Invalid Command Format;
+        stop
+    elseif (Check Duplicate Prefix) then ([Duplicate Prefix])
+        :ParseException: Duplicate Prefix;
+        stop
+    elseif (Parse ID) then ([Invalid ID])
+        :ParseException: Invalid Delivery ID;
+         stop
+    elseif (Parse Note) then ([Invalid Note])
+        :ParseException: Invalid Note;
+        stop
+    else ([Valid ID and Note Provided])
```
Hmm, not sure if we are allowed to specify the conditions in the diamond

zhonghan721 reviewed (commented):
Some general comments

B-enguin committed:
Modify DeliveryCreateNoteActivityDiagram.puml.puml to only include guard conditions

Inline with nus-cs2103-AY2324S1/forum#293, Alternate path conditions should
not be specified in the branch node, and only guard conditions should be used.
Hence, the activity diagram was modified to not use conditions in the branch
node.

B-enguin committed:
Update DeveloperGuide.md following comments

Following the comments given in the PR, DeveloperGuide.md was updated
as suggested by the comments given.

zhonghan721 commented on docs/diagrams/DeliveryCreateNoteSequenceDiagram.puml:
```diff
@@ -0,0 +1,98 @@
+DeliveryCreateNoteCommandParser -[hidden]-> AddressBookParser
+destroy DeliveryCreateNoteCommandParser
+
+AddressBookParser --> LogicManager : d
+deactivate AddressBookParser
+
+AddressBookParser --> LogicManager : d
+deactivate AddressBookParser
```
Is this a duplication?

B-enguin replied:
Yep, good catch, will update accordingly

zhonghan721 reviewed (commented):
Some general comments

B-enguin committed:
Modify DeliveryCreateNoteActivityDiagram.puml to follow formatting

Include conditions for all paths in activity diagram.

Fix duplication of method return for parseCommand.

B-enguin committed:
Merge branch 'master' into 163-update-dg-delivery-create-note

zhonghan721 reviewed (approved):
LGTM
